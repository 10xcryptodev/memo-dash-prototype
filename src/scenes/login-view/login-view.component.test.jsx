import React from 'react'
import { shallow } from 'enzyme'
import LoginViewComponent from './login-view.component'

describe('<LoginViewComponent />', () => {
  let wrapper

  beforeEach(() => {
    const div = document.createElement('div')
    document.body.appendChild(div)
  })

  it('renders without crashing', () => {
    wrapper = shallow(<LoginViewComponent />)
    expect(wrapper).toMatchSnapshot()
  })

  it('renders error message if login failed', () => {
    wrapper = shallow(<LoginViewComponent loginError="LoginError" />)
    expect(wrapper).toMatchSnapshot()
  })
})
