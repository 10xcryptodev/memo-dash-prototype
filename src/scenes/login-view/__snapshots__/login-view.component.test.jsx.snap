// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoginViewComponent /> renders error message if login failed 1`] = `
<Fragment>
  <Menu
    borderless={true}
    color="blue"
    fixed="top"
    size="tiny"
    stackable={true}
  >
    <Container>
      <MenuItem
        as="a"
        header={true}
      >
        <Logo />
        <div
          className="active"
        >
          MemoDash
        </div>
      </MenuItem>
    </Container>
  </Menu>
  <div
    className="login-form"
  >
    <Form
      as="form"
      onSubmit={[Function]}
      size="large"
    >
      <Segment>
        <Header
          as="h2"
          color="blue"
        >
          <Responsive
            getWidth={[Function]}
            maxWidth={767}
            minWidth={320}
          >
            Log-in
          </Responsive>
          <Responsive
            getWidth={[Function]}
            minWidth={768}
          >
            Log-in to your account
          </Responsive>
        </Header>
        <FormInput
          as={[Function]}
          control={[Function]}
          name="username"
          onChange={[Function]}
          placeholder="Username"
          required={true}
          value="alice"
        />
        <br />
        <Button
          as="button"
          color="blue"
          role="button"
          size="large"
        >
          Login
        </Button>
      </Segment>
    </Form>
    <Message
      hidden={false}
      negative={true}
    >
      LoginError
    </Message>
  </div>
</Fragment>
`;

exports[`<LoginViewComponent /> renders without crashing 1`] = `
<Fragment>
  <Menu
    borderless={true}
    color="blue"
    fixed="top"
    size="tiny"
    stackable={true}
  >
    <Container>
      <MenuItem
        as="a"
        header={true}
      >
        <Logo />
        <div
          className="active"
        >
          MemoDash
        </div>
      </MenuItem>
    </Container>
  </Menu>
  <div
    className="login-form"
  >
    <Form
      as="form"
      onSubmit={[Function]}
      size="large"
    >
      <Segment>
        <Header
          as="h2"
          color="blue"
        >
          <Responsive
            getWidth={[Function]}
            maxWidth={767}
            minWidth={320}
          >
            Log-in
          </Responsive>
          <Responsive
            getWidth={[Function]}
            minWidth={768}
          >
            Log-in to your account
          </Responsive>
        </Header>
        <FormInput
          as={[Function]}
          control={[Function]}
          name="username"
          onChange={[Function]}
          placeholder="Username"
          required={true}
          value="alice"
        />
        <br />
        <Button
          as="button"
          color="blue"
          role="button"
          size="large"
        >
          Login
        </Button>
      </Segment>
    </Form>
    <Message
      hidden={true}
      negative={true}
    />
  </div>
</Fragment>
`;
